tests/functional/aws-node-sdk/test/bucket/putBucketLogging.js

```diff
@@ -0,0 +1,159 @@
+    LoggingEnabled: {
+        TargetBucket: targetBucket,
+        TargetPrefix: 'logs/',
+    },
+};
+
+const validLoggingConfigWithGrants = {
+    LoggingEnabled: {
+        TargetBucket: targetBucket,
+        TargetPrefix: 'access-logs/',
+        TargetGrants: [
+            {
+                Grantee: {
+                    Type: 'Group',
+                    URI: 'http://acs.amazonaws.com/groups/s3/LogDelivery',
+                },
+                Permission: 'WRITE',
+            },
+            {
+                Grantee: {
+                    Type: 'Group',
+                    URI: 'http://acs.amazonaws.com/groups/s3/LogDelivery',
+                },
+                Permission: 'READ_ACP',
+            },
+        ],
+    },
+};
+
+describe('PUT bucket logging', () => {
+    withV4(sigCfg => {
+        const bucketUtil = new BucketUtility('default', sigCfg);
+        const s3 = bucketUtil.s3;
+        const otherAccountBucketUtility = new BucketUtility('lisa', {});
+        const otherAccountS3 = otherAccountBucketUtility.s3;
+
+        function _testPutBucketLoggingError(account, config, statusCode, errMsg, cb) {
+            account.putBucketLogging({
+                Bucket: bucketName,
+                BucketLoggingStatus: config,
+            }, err => {
+                assert(err, 'Expected err but found none');
+                assert.strictEqual(err.code, errMsg);
+                assert.strictEqual(err.statusCode, statusCode);
+                cb();
+            });
+        }
+
+        describe('without existing bucket', () => {
+            it('should return NoSuchBucket', done => {
+                _testPutBucketLoggingError(s3, validLoggingConfig, 404, 'NoSuchBucket', done);
+            });
+        });
+
+        describe('with existing bucket', () => {
+            beforeEach(done => {
+                process.stdout.write('Creating buckets\n');
+                return s3.createBucket({ Bucket: bucketName }, err => {
+                    if (err) {
+                        return done(err);
+                    }
+                    return s3.createBucket({ Bucket: targetBucket }, done);
+                });
+            });
+
+            afterEach(done => {
+                process.stdout.write('Deleting buckets\n');
+                bucketUtil.deleteOne(bucketName).then(() => bucketUtil.deleteOne(targetBucket)).then(() => done())
+                    .catch(err => {
+                        if (err && err.code !== 'NoSuchBucket') {
+                            return done(err);
+                        }
+                        return done();
+                    });
+            });
+
+            it('should put bucket logging configuration successfully', done => {
+                s3.putBucketLogging({
+                    Bucket: bucketName,
+                    BucketLoggingStatus: validLoggingConfig,
+                }, err => {
+                    assert.strictEqual(err, null,
```

Comment:
Use ifError if possible
```suggestion
                    assert.ifError(err)
```

Reply:
done